pkg/gds/models/v1/models.go

```diff
@@ -148,6 +148,131 @@ func UpdateCertificateRequestStatus(request *CertificateRequest, state Certifica
 	return nil
 }
 
+// GetReviewNotes returns all of the review notes for a VASP as a map.
+func GetReviewNotes(vasp *pb.VASP) (_ map[string]*ReviewNote, err error) {
+	// If the extra data is nil, return nil (no review notes).
+	if vasp.Extra == nil {
+		return nil, nil
+	}
+
+	// Unmarshal the extra data field on the VASP.
+	extra := &GDSExtraData{}
+	if err = vasp.Extra.UnmarshalTo(extra); err != nil {
+		return nil, err
+	}
+	return extra.ReviewNotes, nil
+}
+
+// CreateReviewNote creates a note on the VASP given the specified note id.
+func CreateReviewNote(vasp *pb.VASP, id string, author string, text string) (err error) {
+	// Validate note id.
+	if id == "" {
+		return errors.New("must specify a valid note id")
+	}
+
+	// Unmarshal previous extra data.
+	extra := &GDSExtraData{}
+	if vasp.Extra != nil {
+		if err = vasp.Extra.UnmarshalTo(extra); err != nil {
+			return fmt.Errorf("could not deserialize previous extra: %s", err)
+		}
+	}
+
+	if extra.ReviewNotes == nil {
+		extra.ReviewNotes = make(map[string]*ReviewNote)
+	}
```

Comment:
If `vasp.Extra` doesn't exist yet, might we get a nil pointer on line 181 because ReviewNotes doesn't exist yet also? I always forget how this works in Go, but I wonder if we need to do something like:

```golang
// Unmarshal previous extra data.
extra := &GDSExtraData{}
if vasp.Extra != nil {
	if err = vasp.Extra.UnmarshalTo(extra); err != nil {
		return fmt.Errorf("could not deserialize previous extra: %s", err)
	}
} else {
        extra.ReviewNotes = make(map[string]*ReviewNote)
}
```

Reply:
I'm pretty sure Go lets us access extra.ReviewNotes, even though it's nil, because we have already instantiated a GDSExtraData on line 174. If we tried to access something inside ReviewNotes we would definitely get an error. I originally ran into a nil pointer exception which is why I added this check and the make.